File: src/db/models/Item.ts
import { ObjectId } from "mongodb";
import { getDatabase } from "../database";
import { input } from "../../utils/inputHelper";

export interface Item  {
  _id?: ObjectId;
  sku: string | null
  item_id:string
  attributes: any
  data: any
}

export const createItem = async (item: Item) => {
  const db = getDatabase();
  const result = await db.collection("mlitems").insertOne(item);
  return result.insertedId ? true : false;
};

export const getAllItems = async (): Promise<Item[]> => {
  const db = getDatabase();
  const items = await db.collection<Item>("mlitems").find().toArray();
  return items;
};

export const itemExistBySku = async (sku: string) => {
  const db = getDatabase();
  const item = await db.collection<Item>("items").findOne({ sku });
  return item ? true : false
};

export const getItemBySku = async (sku: string) => {
  const db = getDatabase();
  const item = await db.collection<Item>("items").findOne({ sku });
  return item
}

export const deleteItemByItemId = async (item_id: string) => {
  const db = getDatabase();
  const item = await db.collection<Item>("mlitems").deleteOne({ item_id });
  return item.deletedCount > 0
}



export const removeDuplicates = async () => {
  try {
    const db = getDatabase();
    const collection = db.collection("mlitems");

    // Identificar duplicados
    const pipeline = [
      {
        $group: {
          _id: {
            field1: "item_id", // Cambia esto a los campos que definen un duplicado
            // field2: "$field2", // Añade más campos si es necesario
          },
          count: { $sum: 1 },
          ids: { $push: "$_id" },
        },
      },
      {
        $match: {
          count: { $gt: 1 },
        },
      },
    ];

    const duplicates = await collection.aggregate(pipeline).toArray();
    console.log("Found duplicates:", duplicates.length);

    await input("press any key.")

    // Eliminar duplicados
    for (const doc of duplicates) {
      const [, ...idsToRemove] = doc.ids;
      await collection.deleteMany({
        _id: { $in: idsToRemove },
      });
    }

    console.log("Duplicates removed");
  } catch (error) {
    console.error("Error removing duplicates:", error);
  } finally {
  }
};

File: py/models/item.py
from db.database import get_database


def create_item(item):
    db = get_database()
    collection = db["products"]
    result = collection.insert_one(item)
    return result.inserted_id


def get_items():
    db = get_database()
    collection = db["products"]
    result = collection.find({}, {'_id': 0})
    return result


def get_item_by_sku(sku):
    db = get_database()
    collection = db["products"]
    result = collection.find_one(
        {'sku': sku, 'state':{"$in":["active", "updated"]}}, {'_id': 0, 'title': 0, 'state': 0})
    return result


def update_price(sku, new_price):
    db = get_database()
    collection = db["products"]
    result = collection.update_one(
        {'sku': sku }, {'$set': {'price': new_price, 'state':"re-updated"}})
    return result.modified_count > 0

def set_revised_state(sku):
    db = get_database()
    collection = db["products"]
    result = collection.update_one(
        {'sku': sku }, {'$set': {'state':"revised"}})
    return result.modified_count > 0